ansible/templates/minigraph_cpg.j2

```diff
@@ -80,20 +80,40 @@
 {% endfor %}
 {% endif %}
 {% if switch_type is defined and switch_type == 'voq' %}
-{% for all_idx in range(all_inbands|length) %}
-{% if voq_inband_ip != all_inbands[all_idx] %}
+{% set voq_peers = dict() %}
+{% for asic_id in range(num_asics) %}
+{% if num_asics == 1 %}
+{% set start_rtr = inventory_hostname %}
+{% else %}
+{% set start_rtr = "ASIC" + asic_id|string %}
+{% endif %}
+{% for a_linecard in all_inbands %}
+{% for idx in range(all_inbands[a_linecard]|length) %}
+{% if voq_inband_ip[asic_id] != all_inbands[a_linecard][idx] %}
       <BGPSession>
-        <StartRouter>{{ inventory_hostname  }}</StartRouter>
-        <StartPeer>{{ voq_inband_ip.split('/')[0] }}</StartPeer>
-        <EndRouter>{{ all_hostnames[all_idx] }}</EndRouter>
-        <EndPeer>{{ all_inbands[all_idx].split('/')[0] }}</EndPeer>
+        <StartRouter>{{ start_rtr }}</StartRouter>
+        <StartPeer>{{ voq_inband_ip[asic_id].split('/')[0] }}</StartPeer>
+{% if all_inbands[a_linecard]|length == 1 %}
+{% set end_rtr = a_linecard %}
+{% else %}
+{% if a_linecard == inventory_hostname %}
+{% set end_rtr = "ASIC" + idx|string %}
+{% else %}
+{% set end_rtr = a_linecard + "-ASIC" + idx|string %}
+{% endif %}
+{% endif %}
+{% set _dummy = voq_peers.update({ all_inbands[a_linecard][idx].split('/')[0] : end_rtr }) %}
```

Comment:
seems like _dummy isn't used anywhere

Reply:
Resolved in commt https://github.com/Azure/sonic-mgmt/pull/3746/commits/c8602e1275e93ac7c2febbabfa543c9725e60bc7